dotcom-rendering/src/web/layouts/ImmersiveLayout.tsx

```diff
@@ -188,16 +188,20 @@ interface Props {
 	format: ArticleFormat;
 }
 
-const decideCaption = (mainMedia: ImageBlockElement | undefined): string => {
+const decideCaption = (mainMedia: FEElement | undefined): string => {
 	const caption = [];
-	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- because sometimes mainMedia isn't an image
-	if (mainMedia?.data?.caption) {
-		caption.push(mainMedia.data.caption);
-	}
 
-	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- because sometimes mainMedia isn't an image
-	if (mainMedia?.displayCredit && mainMedia?.data?.credit) {
-		caption.push(mainMedia.data.credit);
+	if (
+		mainMedia?._type ===
+		'model.dotcomrendering.pageElements.ImageBlockElement'
+	) {
```

Comment:
Can other types of main media have captions? Should we make sure we handle those too?

Reply:
Good question, There are but I think they're handled differently. We pass `hideCaption` to our MainMedia elements, but most of them except for ImageBlockElements ignore that field, so I think we're intentionally only handling captions for ImageBlockElements here.

Previously `decideCaption` only worked with ImageBlockElements too, so I'm hesitant to start adding new behaviour.

Reply:
Very good question, and while indeed not for this PR, I think we should support captions whenever there are ones passed from tools. Like [here](https://www.theguardian.com/world/2023/mar/21/british-airstrikes-mosul-civilian-victims-iraq):

![image](https://user-images.githubusercontent.com/6032869/232091376-ac18ffa9-f0b6-42a6-aec9-68f2a5cbb158.png)

Reply:
We do support them in-body. Modulo https://github.com/guardian/dotcom-rendering/issues/5110.